### PR TITLE
1 command for play/pause/resume

### PR DESCRIPTION
**Pull request by 1C0D**
Only 2 commands available now. Play and Stop.  
Play is assuming 3 functions:  
- Play from the start of the current note  or selection 
   
then  
- Pause
- Resume from where you paused

an updateInfo modal has been added.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -48,26 +71,6 @@ export default class TTSPlugin extends Plugin {
...
-		this.addCommand({
-			id: 'resume-tts-playback',
-			name: 'Resume playback',
-			checkCallback: (checking: boolean) => {
-				if (!checking)
-					this.ttsService.resume();
-				return this.ttsService.isPaused();
-			}
-		});
-
```
Comment (joethei) [inline src/main.ts:70]: Please don't remove existing commands.

File: src/main.ts [hunk 2]
```diff
@@ -2,21 +2,23 @@ import {
...
+import { registerAPI } from "@vanakat/plugin-api";
+import { TTSService } from "./TTSService";
+import { TTSPluginUpdateInfo } from './updateInfo';
 
 
 export default class TTSPlugin extends Plugin {
 	ttsService: TTSService;
 	settings: TTSSettings;
 	statusbar: HTMLElement;
+	paused: boolean = false;
```
Comment (joethei) [inline src/main.ts:16]: The `TTSService` already has a method for this that's more reliable.

File: src/main.ts [hunk 3]
```diff
@@ -25,15 +27,36 @@ export default class TTSPlugin extends Plugin {
...
+			this.settings.savedVersion !== this.manifest.version // new version
+		) {
+			new TTSPluginUpdateInfo(this.app, this).open();
+		}
 
 		this.addCommand({
 			id: 'start-tts-playback',
 			name: 'Start playback',
 			checkCallback: (checking: boolean) => {
-				const markdownView = this.app.workspace.getActiveViewOfType(MarkdownView);
```
Comment (joethei) [inline src/main.ts:33]: Don't change the behavior of an existing command, instead add a new one.

File: src/main.ts [hunk 4]
```diff
@@ -25,15 +27,36 @@ export default class TTSPlugin extends Plugin {
...
+					const markdownView = this.app.workspace.getActiveViewOfType(MarkdownView);
+					if (markdownView) {
+						if (this.ttsService.isSpeaking() && !this.paused) {
+							console.debug("pausing")
+							this.ttsService.pause()
+							this.paused = true
+						}
+						else if (this.paused) {
+							console.debug("resuming")
+							this.ttsService.resume(); this.paused = false
```
Comment (joethei) [inline src/main.ts:51]: Please stick to the code style of the rest of the code.
That means semicolons at the end, and one call per line.

File: src/updateInfo.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            <li>Play from the start of the current note  or selection</li>
+            </ul>
+            <p>then</p>
+            <ul>
+            <li>Pause</li>
+            <li>Resume from where you paused</li>
+            </ul>
+        `;
+        contentEl.createDiv("", (el: HTMLDivElement) => {
+            el.innerHTML = content;
```
Comment (joethei) [inline src/updateInfo.ts:29]: Don't use `innerHTML` whenever possible, even if the HTML is hardcoded.

File: src/updateInfo.ts
Comment (joethei) [inline src/updateInfo.ts:1]: Remove this modal and the setting related to it.
We haven't mentioned it explicitly, but we don't consider these modals good UX in the current implementation.
Having to close multiple modals after each update is not exactly user-friendly.

**Review by joethei — CHANGES_REQUESTED** [review]

**Comment by 1C0D** [comment]
done

**Comment by 1C0D** [comment]
Oops sorry my last commit had an error. Apparently did some ctrl Z...
To be clear, I deleted 2 await because it's not a promise

**Pull request closed without merging**